academy/2-cosmos-concepts/4-messages.md

```diff
@@ -131,80 +141,69 @@ Focus on the messages around the **game creation**. There is no single true way
+  rpc CreateGame(MsgCreateGame)
+    returns (MsgCreateGameResponse);
+}
+```
 
-1. Getting the signer, the `Creator`, of your message:
+After compilation, you register this together with your interface registry so that the module knows to send any `MsgCreateGame` to the `CreateGame` function of the `MsgClient` interface.
 
-    ```go
-    func (msg *MsgCreateGame) GetSigners() []sdk.AccAddress {
-        creator, err := sdk.AccAddressFromBech32(msg.Creator)
-        if err != nil {
-           panic(err)
-       }
-       return []sdk.AccAddress{creator}
-    }
-    ```
+You also define a local `msgServer` that implements the `CreateGame` function with the required actions, such as:
 
-    Where `GetSigners` is [a requirement of `sdk.Msg`](https://github.com/cosmos/cosmos-sdk/blob/1dba673/types/tx_msg.go#L21).
+* Extracting and verifying the players and other values from the message.
+* Creating a new game and saving it to storage.
+* Returning the expected message.
 
-2. Making sure the message's bytes to sign are deterministic:
+**How to proceed with v0.47 or earlier and Ignite**
 
-    ```go
-    func (msg *MsgCreateGame) GetSignBytes() []byte {
-        bz := ModuleCdc.MustMarshalJSON(msg)
-        return sdk.MustSortJSON(bz)
-    }
-    ```
+Thinking from design to implementation, Ignite CLI can help you create these elements, plus the `MsgCreateGame` and `MsgCreateGameResponse` objects, with this command:
 
-3. Adding a callback for your new message type in your module's message handler `x/checkers/handler.go`:
+```sh
+$ ignite scaffold message createGame black red \
+    --module checkers \
+    --response gameIndex
+```
 
-    ```go
-    ...
-    switch msg := msg.(type) {
-        ...
-        case *types.MsgCreateGame:
-            res, err := msgServer.CreateGame(sdk.WrapSDKContext(ctx), msg)
-            return sdk.WrapServiceResult(ctx, res, err)
-        ...
-    }
-    ```
+<HighlightBox type="info">
 
-4. Creating an empty shell of a file (`x/checkers/keeper/msg_server_create_game.go`) for you to include your code, and the response message:
+Ignite CLI creates a variety of other files. See [Run Your Own Cosmos Chain](/hands-on-exercise/1-ignite-cli/index.md) for details, and to make additions to existing files.
 
-    ```go
-    func (k msgServer) CreateGame(goCtx context.Context, msg *types.MsgCreateGame) (*types.MsgCreateGameResponse, error) {
-        ctx := sdk.UnwrapSDKContext(goCtx)
+</HighlightBox>
 
-        // TODO: Handling the message
-        _ = ctx
+_**A sample of things Ignite CLI does for you**_
 
-        return &types.MsgCreateGameResponse{}, nil
-    }
-    ```
+Ignite CLI significantly reduces the amount of work a developer has to do to build an application with the Cosmos SDK. Among others, it:
+
+* Assists with making sure that your message conforms to `sdk.Msg` by adding the boilerplate for `GetSigners` and `GetSignBytes`.
```

Comment:
these are no longer present in 0.50

Reply:
Yes. This update refers to what Ignite does in the v0.47 context (see line 181).

On line 148, we "do the same" but for v0.50:

> The SDK takes care of a lot of things. In particular, by annotating your Protobuf message, you can have it implement `sdk.Msg` after compilation.

Sounds good?

Reply:
ah sorry, yea thats fine